aten/src/ATen/cpu/vec256/vec256_int.h

```diff
@@ -196,7 +196,9 @@ struct Vec256<int32_t> : public Vec256i {
     return _mm256_loadu_si256(reinterpret_cast<const __m256i*>(ptr));
   }
   static Vec256<int32_t> loadu(const void* ptr, int32_t count) {
-    __at_align32__ int32_t tmp_values[size()];
+    // Ensure uninitialized memory does not change the output value
+    // See https://github.com/pytorch/pytorch/issues/32502 for more details
+    __at_align32__ int32_t tmp_values[size()] = {0};
```

Comment:
For consistency with the other initialization in this folder [here](https://github.com/pytorch/pytorch/blob/f72307db4ce2f2862007285ad9c101e6d76683db/aten/src/ATen/cpu/vec256/vec256_int.h#L88-L92) maybe use the for-loop as well?

Reply:
I'm actually planning to remove the for-loops (along with other cleanups) in a followup PR, which is more concise. Is this OK?

Reply:
The main reason we went with the for-loop is because the assembly is slightly better (not sure why) as it gets one instruction instead of two in the gcc we most commonly user: https://godbolt.org/z/KsE59P

Reply:
That's interesting! I've made the changes. Meanwhile I'll ask the gcc community about why this is happening.

Reply:
Would be interested to hear what they think!